### PR TITLE
Integrate motor controller

### PR DESCRIPTION
**Pull request by yagmurdulger**
Prior to this PR, our `move_to_target_node` and `turn_to_target_heading` functions weren't integrated with the electrical `motor_controller` code, and so, all movement was simulated. With these changes:
- Robot class is able to call `spin_motors` function with the desired (clamped) linear and angular velocities as parameters
- The motor controller then converts this information into different voltages that should be applied to the left and right motors for desired movement.
- Additionally, we now initialize `motor_controller` in `Robot` rather than `Mission` which enables the above functionality.

### CODE REVIEW

File: engine/mission.py [hunk 1]
```diff
@@ -63,32 +64,34 @@ def __init__(self, robot, base_station, init_control_mode, grid=Grid(42.444250,
         self.inactive_waypoints = self.grid.get_inactive_waypoints_list()
         self.waypoints_to_visit = deque(self.all_waypoints)
         self.allowed_dist_error = allowed_dist_error
-        if not robot.is_sim:
-            self.gps_serial = serial.Serial('/dev/ttyACM0', 19200, timeout=5) 
-            self.robot_radio_serial = serial.Serial('/dev/ttyS0', 57600) #robot radio 
+        if self.robot.is_sim:
```
Comment (juliehelenek) [inline engine/mission.py:67]: should be if not self.robot.is_sim

File: engine/mission.py [hunk 2]
```diff
@@ -63,32 +64,34 @@ def __init__(self, robot, base_station, init_control_mode, grid=Grid(42.444250,
...
-
     def execute_mission(self, database):
         """
         Activates the main control loop. Depending on the robot's phase, different motion control algorithms are
         activated.
         """
         while self.robot.phase != Phase.COMPLETE:
             if self.robot.phase == Phase.SETUP:
-                self.robot.execute_setup(self.robot_radio_session, self.gps, self.imu, self.motor_controller)
+                self.robot.execute_setup(
```
Comment (juliehelenek) [inline engine/mission.py:93]: is the spacing weird when u open in vs code?

File: engine/robot.py [hunk 3]
```diff
@@ -102,6 +104,9 @@ def __init__(self, x_pos, y_pos, heading, epsilon, max_v, radius, is_sim=True, p
         self.gyro_rotation = [0, 0, 0]  # TEMPORARY
         self.linear_v = 0
         self.angular_v = 0
+        if not self.is_sim:
```
Comment (juliehelenek) [inline engine/robot.py:107]: do we still want to initialize one when we don't have a simulation?

Comment (juliehelenek) [reply engine/robot.py:107]: why are we only initializing one if its not a simulation, but then later in robot.py we only use the motorcontroller if it IS a simulation?

**Review by angelaloh09 — CHANGES_REQUESTED** [review]
some small changes to be made!

**Review by angelaloh09 — APPROVED** [review]
looks good to me

**Pull request merged**